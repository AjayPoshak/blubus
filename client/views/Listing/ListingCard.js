import React from 'react';
import { CardWrapper, CardHeader, CompanyName, Currency, Timings, BusIcon, Timer, TimerWrapper } from './style';

const ListingCard = props => (
	<CardWrapper>
		<CardHeader>
			<CompanyName>Paulo Travels</CompanyName>
			<Currency>70 USD</Currency>
		</CardHeader>
		<Timings>
			<p>1730</p>
			<BusIcon alt="traveling-bus" src="http://res.cloudinary.com/ddbxa4afa/image/upload/v1528048630/blubus/traveling-bus.svg" />
			<p>0600</p>
			<TimerWrapper>
				<Timer alt="timer" src="http://res.cloudinary.com/ddbxa4afa/image/upload/v1528048884/blubus/timer.svg" />
				<p>18 hrs</p>
			</TimerWrapper>
		</Timings>
	</CardWrapper>
);

export default ListingCard;

