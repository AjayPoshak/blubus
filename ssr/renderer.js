export const createScriptTag = script => `<script src=${script}></script>`;

const renderFullPage = (html, preloadedState, styles, bundles) => `<!DOCTYPE html>
	<html lang="en">
	
	<head>
		<meta charset="UTF-8">
		<title>Bus Booking in Africa</title>
		<meta http-equiv="X-UA-Compatible" content="ie=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<!-- multiple icons for IE -->
		<meta name="theme-color" content="#2396b0">
		<meta name="msapplication-square310x310logo" content="icon_largetile.png">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
		<meta name="description" content="online bus booking service in Rwanda, 
		book bus online from anywhere">
		<!-- icon in the highest resolution we need it for -->
		<!-- reuse same icon for Safari -->
		<link rel="manifest" href="../public/manifest.json">
		<link rel="apple-touch-icon" sizes="57x57" href="/apple-icon-57x57.png">
		<link rel="apple-touch-icon" sizes="60x60" href="/apple-icon-60x60.png">
		<link rel="apple-touch-icon" sizes="72x72" href="/apple-icon-72x72.png">
		<link rel="apple-touch-icon" sizes="76x76" href="/apple-icon-76x76.png">
		<link rel="icon" sizes="192x192" href="/public/images/BluBus_192x192.png">
		<link rel="icon" type="image/png" sizes="96x96" href="/favicon-96x96.png">
		<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
		<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
		<link rel="apple-touch-icon" sizes="114x114" href="/apple-icon-114x114.png">
		<link rel="apple-touch-icon" sizes="120x120" href="/apple-icon-120x120.png">
		<link rel="apple-touch-icon" sizes="144x144" href="/apple-icon-144x144.png">
		<link rel="apple-touch-icon" sizes="152x152" href="/apple-icon-152x152.png">
		<link rel="apple-touch-icon" sizes="180x180" href="/apple-icon-180x180.png">
		<link rel="apple-touch-startup-image" href="/public/images/BluBus_192x192.png">
		<link rel="icon" type="image/png" sizes="192x192"  href="/android-icon-192x192.png">
		<meta name="msapplication-TileColor" content="#ffffff">
		<meta name="msapplication-TileImage" content="/ms-icon-144x144.png">
		${styles}
	</head>
	
	<body>
		<div id="root">${html}</div>
	</body>
	<link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
	<script src=https://cdn.ravenjs.com/3.24.0/raven.min.js
 crossorigin=anonymous></script>
	<script>
		window.__PRELOADED_STATE__ = ${JSON.stringify(preloadedState).replace(/</g, '\\u003c')}
	</script>
	<script src=${bundles.manifest.js}></script>
	<script src=${bundles.main.js}></script>
	<script src=${bundles.vendors.js}></script>
	<script>
		if('serviceWorker' in navigator) {
			window.addEventListener('load', () => {
				navigator.serviceWorker.register('./service-worker.js')
			})
		}
	</script>
	<!-- Third Party CSS for calendar module -->
	<link rel="stylesheet" href="/public/rc-calendar.css">
	</html>`;
export default renderFullPage;
