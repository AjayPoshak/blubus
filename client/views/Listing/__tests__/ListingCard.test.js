/* eslint function-paren-newline: ["error", "never"] */
import React from 'react';
import { render, shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import ListingCard from '../ListingCard';

// describe('has click handler', () => {});
function renderListingCard(props = {}) {
	return render(<ListingCard {...props} />);
}
describe('Testing ListingCard Component', () => {
	it('has company name', () => {
		const wrapper = renderListingCard({ companyName: 'PR Travels' });
		expect(wrapper.find('h2').text()).toBe('PR Travels');
	});

	it('has currency', () => {
		const wrapper = renderListingCard({ currency: '$' });
		expect(wrapper.find('header p').text()).toBe(' $');
	});

	it('has price', () => {
		const wrapper = renderListingCard({ price: '210' });
		expect(wrapper.find('header p').text()).toBe('210 ');
	});

	it('has arrival time', () => {
		const wrapper = renderListingCard({ arrivalTime: '12:30 PM' });
		expect(
			wrapper
				.find('article p')
				.first()
				.text()
		).toBe('12:30 PM');
	});

	it('has departure time', () => {
		const wrapper = renderListingCard({ departureTime: '1:30 PM' });
		expect(wrapper.find('article > p').text()).toBe('1:30 PM');
	});

	it('has travel time', () => {
		const wrapper = renderListingCard({ travelTime: '11' });
		expect(
			wrapper
				.find('article p')
				.last()
				.text()
		).toBe('11 hours');
	});

	it('onClick handler has been called', () => {
		// const clickHandler = jest.fn();
		// const wrapper = mount(<ListingCard />, { clickHandler });
		// expect(wrapper.find('section').length).toBe(1);
		// wrapper.find('section').simulate('click');
		// expect(clickHandler.mock.calls.length).toBe(1);
	});

	// Compare snapshot
	it('compare snapshot for UI changes', () => {
		const wrapper = renderer
			.create(
				mount(
					<ListingCard
						companyName="Pawan Tours"
						currency="#"
						price="511"
						arrivalTime="11:00 AM"
						departureTime="12:30 PM"
						travelTime="9 hours"
					/>
				)
			)
			.toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
