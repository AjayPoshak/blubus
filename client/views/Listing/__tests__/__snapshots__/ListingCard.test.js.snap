// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ListingCard Component compare snapshot for UI changes 1`] = `
<section
  className="style__CardWrapper-s1qs7nmh-1 cPWONS"
  onClick={[Function]}
>
  <header
    className="style__CardHeader-s1qs7nmh-4 hMmTCZ"
  >
    <h2
      className="style__CompanyName-s1qs7nmh-5 gDLqQP"
    >
      Pawan Tours
    </h2>
    <p
      className="style__Currency-s1qs7nmh-6 kitmCW"
    >
      511
       
      #
    </p>
  </header>
  <article
    className="style__Timings-s1qs7nmh-7 keHeXf"
  >
    <p>
      11:00 AM
    </p>
    <img
      alt="traveling-bus"
      className="style__BusIcon-s1qs7nmh-9 ilmrMz"
      src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528048630/blubus/traveling-bus.svg"
    />
    <p>
      12:30 PM
    </p>
    <div
      className="style__TimerWrapper-s1qs7nmh-8 gGJsSN"
    >
      <img
        alt="timer"
        className="style__Timer-s1qs7nmh-10 cmfDyD"
        src="https://res.cloudinary.com/ddbxa4afa/image/upload/v1528048884/blubus/timer.svg"
      />
      <p>
        9 hours
         hours
      </p>
    </div>
  </article>
</section>
`;
